libs/platform/src/lib/components/menu-button/menu-button.component.ts

```diff
@@ -0,0 +1,75 @@
+import { Component, ChangeDetectionStrategy, OnInit, Input, Output, EventEmitter } from '@angular/core';
+import { InputSize } from './../form/form-control';
+import { ButtonType } from '../button/button.component';
+import { ButtonOptions } from '../button/button.component';
+
+@Component({
+    selector: 'fdp-menu-button',
+    templateUrl: './menu-button.component.html',
+    styleUrls: ['./menu-button.component.scss'],
+    changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class MenuButtonComponent implements OnInit {
+    /** Id for menu-button. */
+    @Input()
+    Id: string;
+
+    /** Option to make to menu-button compact. */
+    @Input()
+    size: InputSize = 'cozy';
+
+    /** Option to truncate content of the menu-button based on width. */
+    @Input()
+    width: string;
+
+    /** The icon to include in the menu-button */
+    @Input()
+    icon: string;
+
+    /** Menu-Button is disabled on true. */
+    @Input()
+    disabled: boolean;
+
+    /** The type of the button. Types include 'standard', 'positive', 'medium', and 'negative'.
+     * Leave empty for default (Action button).'*/
+    @Input()
+    type: ButtonType;
+
+    /** Button options.  Options include 'emphasized' and 'light'. Leave empty for default.' */
+    @Input()
+    options: ButtonOptions;
+
+    /** Event sent when menu-button is clicked */
+    @Output()
+    click: EventEmitter<MouseEvent | KeyboardEvent | TouchEvent> = new EventEmitter();
+
+    /** add 'menu' as button option */
+    /** @hidden */
+    menuOptions: string[] = new Array();
+
+    _iscompact: boolean = false;
```

Comment:
In this case you need to declare this as `private` and use `\** @hidden *\`

Reply:
private variable will not be visible in template. hence can't be used in template.

Reply:
Then why do you prefix it with `_` ? Usually when we see `_varName` we assume it's private

Reply:
Also needs TypeDocs for the API